interfaces/repo_test.go

```diff
@@ -139,26 +185,26 @@ func (s *RepositorySuite) TestAddPlugClash(c *C) {
 	err = s.testRepo.AddPlug(s.plug)
 	c.Assert(err, ErrorMatches, `cannot add plug, snap "provider" already has plug "plug"`)
 	c.Assert(s.testRepo.AllPlugs(""), HasLen, 1)
-	c.Assert(s.testRepo.Plug(s.plug.Snap, s.plug.Name), DeepEquals, s.plug)
+	c.Assert(s.testRepo.Plug(s.plug.Snap.Name, s.plug.Name), DeepEquals, s.plug)
 }
 
 func (s *RepositorySuite) TestAddPlugFailsWithInvalidSnapName(c *C) {
-	plug := &Plug{
-		Snap:      "bad-snap-",
-		Name:      "name",
-		Interface: "interface",
-	}
+	plug := plugFromYaml(c, "name", []byte(`
```

Comment:
I think it is important to decouple parsing from validation so that invalid data can still be represented.
I could implement recursive validation of snap.Info and move the logic there.

Still I'd probably rather add this to the TODO pile and land this without going there yet, I'd like to use this branch as a basis writing security files and not go down all the rabbit holes I see along the way.